### DISCUSSION TITLE
The strange 'COPY' command in docker file in Dockerfile

### DISCUSSION THREAD

**Discussion by naddika** [plausible/analytics#2131] [Anything Else]
In `analytics` in the Dockerfile there're numerous command like these:

```
COPY mix.exs ./
COPY mix.lock ./
```
and

```
COPY assets/package.json assets/package-lock.json ./assets/
```

and

```
COPY assets ./assets
COPY tracker ./tracker
COPY config ./config
[.........]
```

and so on.

Where's is it copying files? Where *from* and where *to*? 

To me it appears that it's copying a file in the current directory ... **to the precisely the same location**: to current directory and under the *same name*!

How's that? 
How does it work? What's the idea here?

---

I've even checked it in the terminal on my local computer (without Docker) and indeed I got an error:

```
cd analytics
cp assets/package.json assets/package-lock.json ./assets/
cp: 'assets/package.json' and './assets/package.json' are the same file
```
and or single file:

```
cd analytics
cp assets/package.json ./assets/
cp: 'assets/package.json' and './assets/package.json' are the same file
```

**Comment by Vigasaurus** [comment]
https://docs.docker.com/engine/reference/builder/#copy

It's copying from the build directory **into** the image's filesystem, not into the same directory it started in (just one within the docker image filesystem with the same path).
